📝 Add plot to show downloads datasets

tkoyama010 opened the pull request:
### Overview

<!-- Please insert a high-level description of this pull request here. -->
This PR add plots to show the downloads datasets.

<!-- Be sure to link other PRs or issues that relate to this PR here. --> 

<!-- If this fully addresses an issue, please use the keyword `resolves` in front of that issue number. -->


### Details

![image](https://user-images.githubusercontent.com/7513610/142721540-e0a1d590-a66a-4f78-99aa-92be86dcac8a.png)

akaszynski commented on pyvista/examples/downloads.py:
```diff
@@ -3208,7 +3566,10 @@ def download_lucy(load=True):  # pragma: no cover
-    >>> dataset = examples.download_lucy()  # doctest:+SKIP
+    >>> dataset = examples.download_lucy()
+    >>> dataset.plot(cpos="iso")
+    >>> examples.delete_downloads()
```
Do we need to delete the downloads?

tkoyama010 replied:
No. I will delete them.

akaszynski commented on pyvista/examples/downloads.py:
```diff
@@ -3208,7 +3566,10 @@ def download_lucy(load=True):  # pragma: no cover
     >>> from pyvista import examples
-    >>> dataset = examples.download_lucy()  # doctest:+SKIP
+    >>> dataset = examples.download_lucy()
+    >>> dataset.plot(cpos="iso")
```
`cpos="iso"` is default, we can probably remove this.

akaszynski commented on pyvista/examples/downloads.py:
```diff
@@ -2901,7 +3217,16 @@ def download_mars_jpg():  # pragma: no cover
+    >>> import pyvista as pv
+    >>> pl = pv.Plotter()
+    >>> dataset = examples.download_mars_jpg()
+    >>> import pyvista as pv
+    >>> pl = pv.Plotter()
+    >>> dataset = examples.download_mars_jpg()
```
```suggestion
```

akaszynski commented:
Overall I think this is a good PR, provided it doesn't affect the stability of our build and doc check.  Can we also add in a cache to the examples directory as we've done in the other PRs?  This way we won't download the examples each time and this will make it faster and more stable.

tkoyama010 commented:
@akaszynski  Thanks. I deleted the ``examples.delete_downloads()``. It will add in a cache to the examples directory.

tkoyama010 commented on pyvista/examples/downloads.py:
```diff
@@ -2918,7 +3030,14 @@ def download_stars_jpg():  # pragma: no cover
+    >>> import pyvista as pv
+    >>> pl = pv.Plotter()
+    >>> dataset = examples.download_stars_jpg()
+    >>> import pyvista as pv
+    >>> pl = pv.Plotter()
+    >>> dataset = examples.download_stars_jpg()
```
```suggestion
```

akaszynski commented:
> It will add in a cache to the examples directory.

See: https://github.com/actions/cache

tkoyama010 commented:
I'm sorry for my lack of understanding. I will try to add it.

adeak commented:
I've been contemplating something that tangentially relates to this PR.

Some example datasets come with notes [such as](https://graphics.stanford.edu/data/3Dscanrep/)
> As you browse this repository and think about how you might use our 3D models and range datasets, please remember that several of these artifacts have religious or cultural significance. Aside from the buddha, which is a religious symbol revered by hundreds of millions of people, the dragon is a symbol of Chinese culture, the Thai statue contains elements of religious significance to Hindus, and Lucy is a Christian angel; statues like her are commonly seen in Italian churches. Keep your renderings and other uses of these particular models in good taste. Don't animate or morph them, don't apply Boolean operators to them, and don't simulate nasty things happening to them (like breaking, exploding, melting, etc.). Choose another model for these sorts of experiments. (You can do anything you want to the Stanford bunny or the armadillo.) 

I wonder if we have a responsibility to echo such warnings in our docstrings, since the examples will now be fully documented complete with figures.

tkoyama010 commented on .github/workflows/testing-and-deployment.yml:
```diff
@@ -81,6 +82,13 @@ jobs:
+          path: ~/.local/share/pyvista/examples
+          key: Examples-${{ hashFiles('*') }}
+          restore-keys: |
+            Examples-${{ hashFiles('*') }}
```
```suggestion
            Examples-
```

tkoyama010 commented on .github/workflows/testing-and-deployment.yml:
```diff
@@ -145,6 +154,13 @@ jobs:
+          path: ~/.local/share/pyvista/examples
+          key: Examples-${{ hashFiles('*') }}
+          restore-keys: |
+            Examples-${{ hashFiles('*') }}
```
```suggestion
            Examples-
```

tkoyama010 commented on .github/workflows/style_docstr.yml:
```diff
@@ -55,6 +55,13 @@ jobs:
+          path: ~/.local/share/pyvista/examples
+          key: Examples-${{ hashFiles('*') }}
+          restore-keys: |
+            Examples-${{ hashFiles('*') }}
```
```suggestion
            Examples-
```

banesullivan commented:
> don't simulate nasty things happening to them (like breaking, exploding, melting, etc.). Choose another model for these sorts of experiments. (You can do anything you want to the Stanford bunny or the armadillo.)

I would really prefer not to see an exploding bunny in our forums or examples

tkoyama010 commented:
> See: https://github.com/actions/cache
@akaszynski I added cache to GitHub Actions. It seems working.

> I wonder if we have a responsibility to echo such warnings in our docstrings, since the examples will now be fully documented complete with figures.
@adeak @banesullivan I added warning in document about the respect for model and using poor bunny and armadillo.

akaszynski commented:
Please wait to review. I'm improving the plots of the examples a bit to make them look nicer:

![image](https://user-images.githubusercontent.com/11981631/146679958-0c903389-948d-4c0e-b64a-b7de77d60133.png)

tkoyama010 commented:
> bit to make them look nicer:

It is much 😍

akaszynski reviewed (approved):
This is good to merge when the checks are green.

tkoyama010 reviewed (commented):
@akaszynski Thanks.

adeak commented on pyvista/examples/downloads.py:
```diff
@@ -2899,8 +3129,45 @@ def download_mars_jpg():  # pragma: no cover
+    >>> mars_tex = pyvista.read_texture(mars_jpg)
+    >>> stars_jpg = examples.download_stars_jpg()
+
+    Create a sphere mesh and compute the texture coordinates.
+
+    >>> sphere = pyvista.Sphere(radius=1, theta_resolution=120, phi_resolution=120,
+    ...                         start_theta=270.001, end_theta=270)
+    >>> sphere.active_t_coords = numpy.zeros((sphere.points.shape[0], 2))
+    >>> sphere.active_t_coords[:, 0] = 0.5 + np.arctan2(-sphere.points[:, 0],
+    ...                                                 sphere.points[:, 1])/(2 * math.pi)
+    >>> sphere.active_t_coords[:, 1] = 0.5 + np.arcsin(sphere.points[:, 2]) / math.pi
```
I can see a few issues in these last few unreviewed commits, I'll fix them in a follow-up. @akaszynski quick question here: why didn't you use our `texture_map_to_sphere` helper? https://docs.pyvista.org/examples/02-plot/texture.html#spherical-texture-coordinates
